tests/Integration/Adapter/ConfigurationTest.php

```diff
@@ -0,0 +1,415 @@
+        yield [
+            [],
+            [
+                'key' => 'does_not_exist',
+                'default' => 'default_value',
+                'shopConstraint' => ShopConstraint::shopGroup(1),
+            ],
+            'default_value',
+        ];
+        // try to get a non existing value for single shop, get default value instead
+        yield [
+            [],
+            [
+                'key' => 'does_not_exist',
+                'default' => 'default_value',
+                'shopConstraint' => ShopConstraint::shop(2),
+            ],
+            'default_value',
+        ];
+        // get value for a group shop, inherited from all shop
+        yield [
+            [
+                'key' => 'all_shop_key_1',
+                'value' => 'all_shop_value_1',
+                'shopConstraint' => ShopConstraint::allShops(),
+            ],
+            [
+                'key' => 'all_shop_key_1',
+                'default' => false,
+                'shopConstraint' => ShopConstraint::shopGroup(1),
+            ],
+            'all_shop_value_1',
+        ];
+        // get value for shop 2, inherited from parent group shop
+        yield [
+            [
+                'key' => 'parent_group_key_1',
+                'value' => 'parent_group_value',
+                'shopConstraint' => ShopConstraint::shopGroup(1),
+            ],
+            [
+                'key' => 'parent_group_key_1',
+                'default' => false,
+                'shopConstraint' => ShopConstraint::shop(2),
+            ],
+            'parent_group_value',
+        ];
+    }
+
+    /**
+     * @return Generator
+     */
+    public function getWithStrictParameterProvider(): Generator
+    {
+        // try getting a non existing value for a aingle shop, with is strict = true => should not inherit from parent group
+        yield [
+            [
+                'key' => 'parent_group_key_2',
+                'value' => 'parent_group_value_2',
+                'shopConstraint' => ShopConstraint::shopGroup(1),
+            ],
+            [
+                'key' => 'parent_group_key_2',
+                'default' => false,
+                'shopConstraint' => ShopConstraint::shop(2, true),
+            ],
+            null,
+        ];
+        // try getting a non existing value for a group, with is strict = true => should not inherit from all shop
+        yield [
+            [
+                'key' => 'all_shop_key_1',
+                'value' => 'all_shop_value_1',
+                'shopConstraint' => ShopConstraint::allShops(),
+            ],
+            [
+                'key' => 'all_shop_key_1',
+                'default' => false,
+                'shopConstraint' => ShopConstraint::shopGroup(1, true),
+            ],
+            null,
+        ];
+        // try getting a non existing value for all shop, with is strict = true => should return null
```

Comment:
You should maybe also test cases where it works ^^ Just to make sure it doesn't always return null 😛